addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/BridgeHandler.java

```diff
@@ -0,0 +1,219 @@
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the bridge handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No channels - nothing to do
+    }
+
+    /**
+     * Start the job refreshing the handler
+     */
+    private void startAutomaticRefresh(BridgeConfiguration config) {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                try {
+
+                    checkConnection();
+
+                } catch (Exception e) {
+                    logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            };
+
+            int delay = config.refresh.intValue();
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    protected BridgeConfiguration getBridgeConfiguration() {
+
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        return config;
+
+    }
+
+    private void checkConnection() throws Exception {
+
+        String errormsg = null;
+        String response1 = null;
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        String hostname = config.hostname;
+
+        String url = "http://" + hostname + ":9763/services/Version.VersionHttpEndpoint";
+
+        URL obj;
+
+        obj = this.validateCertificate(url);
+
+        if (obj == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Certificate validation failed");
+
+        } else {
+
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+
+            con.setRequestMethod("GET");
+            try {
+
+                BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+                String inputLine;
+                StringBuffer response = new StringBuffer();
+
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                in.close();
+
+                response1 = response.toString();
+                logger.debug(response1);
+
+                if (response1.equals("[]")) {
+                    errormsg = "empty response from wso2 iots";
+                    logger.warn(errormsg);
+                }
+            } catch (Exception e) {
+                errormsg = e.getMessage();
+                logger.warn(errormsg);
+            }
+
+            if (errormsg == null) {
+                updateStatus(ThingStatus.ONLINE);
+
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errormsg);
+
```

Comment:
Do no use these additional new lines.

It would be great if you could also review the report of [the Static code analysis](https://docs.openhab.org/developers/development/bindings.html#static-code-analysis), this also saves me some time while reviewing.